quipucords/tests/scanner/network/test_inspect_callback.py

```diff
@@ -0,0 +1,299 @@
+        """Test InspectCallback.event_callback in a unreachable event."""
+        callback = InspectCallback(mocker.Mock())
+        patched_task_on_unreachable = mocker.patch.object(
+            InspectCallback, "task_on_unreachable", wraps=callback.task_on_unreachable
+        )
+        callback.event_callback(event_unreachable)
+        patched_task_on_unreachable.assert_called_once_with(event_unreachable)
+        # this host is hardcoded on the fixture
+        assert "127.0.0.1" in callback._unreachable_hosts
+
+    @pytest.mark.parametrize(
+        "facts,is_unreachable,expected_status",
+        (
+            ({"host_done": True}, False, SystemInspectionResult.SUCCESS),
+            # maybe we should consider the next one successful
+            ({"host_done": True}, True, SystemInspectionResult.UNREACHABLE),
+            ({"pineapple": "abacaxi"}, True, SystemInspectionResult.UNREACHABLE),
+            ({"strawberry": "morango"}, False, SystemInspectionResult.FAILED),
+            ({"host_done": "some string"}, False, SystemInspectionResult.FAILED),
+            ({"host_done": False}, False, SystemInspectionResult.FAILED),
+        ),
+    )
+    def test_results(self, mocker, facts, is_unreachable, expected_status):
+        """Test InspectCallback.iter_results."""
+        callback = InspectCallback(mocker.Mock())
+        callback._ansible_facts = {"host": facts}
+        if is_unreachable:
+            callback._unreachable_hosts.add("host")
+        results = next(callback.iter_results())
+        assert results.facts == facts
+        assert results.status == expected_status
+
+
+@pytest.fixture
+def local_inventory(tmp_path):
+    """Create a inventory for ansible to connect with localhost."""
+    inventory = {"all": {"hosts": {"localhost": {"ansible_connection": "local"}}}}
+    inventory_file = tmp_path / "inventory.yml"
+    contents = yaml.dump(
+        inventory, allow_unicode=True, default_flow_style=False, Dumper=AnsibleDumper
+    )
+    inventory_file.write_text(contents)
+    return str(inventory_file)
+
+
+@pytest.mark.integration
+def test_inspect_callback_with_inspect_playbook(mocker, settings, local_inventory):
+    """Smoketest InspectCallback integration with ansible_runner."""
+    playbook_path = str(settings.BASE_DIR / "scanner/network/runner/inspect.yml")
+    callback = InspectCallback(mocker.Mock())
+
+    runner_obj = ansible_runner.run(
+        inventory=local_inventory,
+        event_handler=callback.event_callback,
+        playbook=playbook_path,
+        extravars=ScanOptions.get_default_extra_vars(),
+    )
+    assert runner_obj.status == "successful"
+    all_results = list(callback.iter_results())
+    assert len(all_results) == 1
+    results = all_results[0]
+    assert results.host == "localhost"
+    assert results.status == SystemInspectionResult.SUCCESS
+    # spot check at least one fact
+    assert results.facts["cpu_count"] == str(os.cpu_count())
```

Comment:
given a recent feedback, I expect this to work on mac, but I'd like someone to please run it to be sure @infinitewarp or @abellotti

Reply:
Unfortunately, this test (but _only_ this test) is failing on my Mac. 😕 Details in Slack thread here: https://redhat-internal.slack.com/archives/C02QSNF1UKE/p1707410433413239?thread_ts=1707317786.877279&cid=C02QSNF1UKE